### DISCUSSION TITLE
Will Remix load JS for routes in a switch?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2948] [Q&A]
Let's say I have the following in a route

```javascript
export default function() {
     const { type } = useLoaderData<{ type: string}>();
    switch(type):
       case "PDF":
          return <ExpensivePDFComponent />
     case "EPUB":
         return <ExpensiveEPUBComponent />
}
```

I happen to know that the loader will not suddenly return a different type. Of course, Remix doesn't know that.
My question is: Let's say the loader returns "pdf", upon hydration, will Remix also load the Javascript for `ExpensiveEPUBComponent`?

**Comment by sergiodxa** [comment]
It will, because you are creating a route with two components which may or may not be rendered but could be rendered.

You'll have to lazy load the component if you want to ensure they are not imported. or create different routes for PDF and EPUB so they have different bundles.
